ace-indexer/src/main/java/edu/isi/vista/aceIndexer/IndexAceWithElasticSearch.java

```diff
@@ -0,0 +1,262 @@
+            String lang,
+            double fractionDocAllowToFail,
+            int sentenceLimit) throws IOException {
+
+        for (List<Path> articlePaths : iterator) {
+            final BulkRequest bulkRequest = new BulkRequest();
+            for (Path articlePath : articlePaths) {
+                Article article = articleFromPath(articlePath);
+                if (article.failed()) { // error occurred
+                    indexFailed += 1;
+                    if ((double)indexFailed/(double)(totalDoc) > fractionDocAllowToFail) {
+                        throw new RuntimeException("Failed documents exceeded threshold");
+                    }
+                } else if (article.getSegments() > sentenceLimit) {
+                    indexFailed += 1;
+                    log.warn("Document not indexed because it exceeded the size limit of {}: {}, {}",
+                            sentenceLimit , article.getSegments(), article.getId());
+                } else {
+                    XContentBuilder sourceBuilder =
+                            buildSourceObject(article, lang, "", new Date().toString(), "");
+                    bulkRequest.add(
+                            new IndexRequest(indexName, "texts", article.getId()).source(sourceBuilder));
+                }
+                totalDoc += 1;
+            }
+            if (bulkRequest.numberOfActions() > 0) {
+                BulkResponse bulkResponse = client.bulk(bulkRequest, RequestOptions.DEFAULT);
+                if (bulkResponse.hasFailures()) {
+                    throw new RuntimeException(bulkResponse.buildFailureMessage());
+                }
+            }
+
+            if (maxDocumentsToIndex.isPresent() && totalDoc >= maxDocumentsToIndex.getAsInt()) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private static XContentBuilder buildSourceObject(
+            Article article, String language, String source, String timestamp, String uri)
+            throws IOException {
+
+        return XContentFactory.jsonBuilder()
+                .startObject()
+                .startObject("doc")
+                .field("text", article.getText())
+                .endObject()
+                .startObject("metadata")
+                .field("id", article.getId())
+                .field("language", language)
+                .field("source", source)
+                .field("timestamp", timestamp)
+                .field("uri", uri)
+                .endObject()
+                .endObject();
+    }
+
+    private static Article articleFromPath(Path filePath) throws IOException{
+
+        String docString = new String(Files.readAllBytes(filePath));
+        // DOC ID marker
+        Pattern ACE_DOC_ID_PATTERN = Pattern.compile("<DOCID> (.*?) <.DOCID>");
+
+        Matcher m =
+                ACE_DOC_ID_PATTERN.matcher(
+                        docString.substring(0, Math.min(120, docString.length())));
```

Comment:
Why the `Math.min`?

Reply:
This is also in the existing gigaword indexer. I believe it is for the edge case where the entire document is less that 120 characters long. To prevent a `StringIndexOutOfBoundsException` from `substring()`